src/Rules/PHPUnit/MockMethodCallRule.php

```diff
@@ -81,7 +80,7 @@ public function processNode(Node $node, Scope $scope): array
 
 			$mockClass = $type->getTypes()[0];
 
-			if (!($mockClass instanceof ObjectType) || $mockClass->hasMethod($method)->yes()) {
+			if (!$mockClass->isObject()->yes() || $mockClass->hasMethod($method)->yes()) {
```

Comment:
The thing with `GenericObjectType` above can be rewritten using getTemplateType (https://apiref.phpstan.org/1.10.x/source-src.Type.Type.html#209-225): `$type->getTemplateType(InvocationMocker::class, 'TMockedClass')`

The part with `$mockClass->getClassName()` should be replaced with `$mockClass->getObjectClassNames()`.